### ISSUE TITLE
let_it_be vs global_fixtures / ActiveSupport::TestCase.fixtures

### ISSUE THREAD

**Issue by jdelStrother** [test-prof/test-prof#206]
### Tell us about your environment

**Ruby Version:** 2.7.2

**Framework Version (RSpec, Minitest, FactoryGirl, Rails, whatever):** RSpec 3.10, Rails 6.0.3

**TestProf Version:** 1.0.1

### What did you do?

Created some fixtures (eg accounts.yml) and specified them in RSpec's `config.global_fixtures` (which calls ActiveSupport::TestCase.fixtures under the hood).
Created 2 files for rspec to test (eg spec/models/post_spec.rb, spec/models/comment_spec.rb)
Created a model in spec/models/post_spec.rb - `let_it_be { create(:post) }`
Use one of the global fixtures in spec/models/comment_spec.rb (eg `account(:admin)`)

### What did you expect to happen?

My global fixtures are loaded and available throughout the entire spec run.
The `let_it_be` model I created in post_spec.rb is loaded and available within the post_spec example group.

### What actually happened?

My global fixtures are loaded for post_spec, but seem to be rolled back at the end and so unavailable for comment_spec.
If I change `let_it_be` to a regular old `let`, it seems that the global fixtures are created before the spec's transaction, so continue to exist after post_spec's rollback.

Does that make any sense? Is this a known limitation? I can't share the code for this project, but let me know if you need me to create a sample project for reproducing it

**Comment by palkan** [comment]
> Does that make any sense? Is this a known limitation?

Yes, I'm aware of this (maybe, not exactly, but similar) issue. Try enabling fixtures within `before_all` globally:

```ruby
TestProf::BeforeAll.configure do |config|
  config.setup_fixtures = true
end
```

See https://test-prof.evilmartians.io/#/recipes/before_all?id=using-rails-fixtures-experimental

**Comment by jdelStrother** [comment]
Thanks! Yep, that setting fixes things, looks like the fixtures now get created before the savepoint that's wrapping the rest of the spec.

**Comment by palkan** [comment]
> looks like the fixtures now get created before the savepoint that's wrapping the rest of the spec.

Exactly.

**Comment by jdelStrother** [comment]
Oh, one caveat: it breaks in Rails 6.1.  TestProf uses `config` here - https://github.com/test-prof/test-prof/blob/876b1d2eccc13ccc9304dd7af9117c98ec32c422/lib/test_prof/before_all/adapters/active_record.rb#L30 - which has been removed in Rails 6.1 (https://github.com/rails/rails/commit/f22dd39cb2adf85d3deeca61f9465206f7bd8df3).

I've monkeypatched around it for now with

```ruby
module ActiveRecord::TestFixtures
  def config
    ActiveRecord::Base
  end
end
```

**Comment by palkan** [comment]
Thanks for catching this!

Pushed a fix to master, could you please try it?

**Comment by jdelStrother** [comment]
Yep, looks good!

**Comment by jdelStrother** [comment]
EDIT: wait, spoke too soon. Let me investigate something....

**Comment by jdelStrother** [comment]
Right, sorry about that. The fix on master fails with:
```
       uninitialized constant TestProf::BeforeAll::Adapters::ActiveRecord::Base
       Did you mean?  Base64
```

I think you want this instead:

```diff
          def setup_fixtures(test_object)
            test_object.instance_eval do
              @@already_loaded_fixtures ||= {}
              @fixture_cache ||= {}
-             config = ActiveRecord::Base
+             config = ::ActiveRecord::Base

              if @@already_loaded_fixtures[self.class]
                @loaded_fixtures = @@already_loaded_fixtures[self.class]
```

**Comment by palkan** [comment]
Thanks! Fixed.

**Comment by palkan** [comment]
Hey @jdelStrother! Could you please try the master version and verify that it works for you? (So, I'll be able to release the latest fix).

**Comment by jdelStrother** [comment]
Yep, seems good!
